File: RelationExample/RelationExample/Quadrangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelationExample
{
    class Quadrangle
    {
        protected float sideLengthA;
        protected float sideLengthB;
        protected float sideLengthC;
        protected float sideLengthD;

        public Quadrangle(float sideLengthA, float sideLengthB, float sideLengthC, float sideLengthD)
        {
            this.sideLengthA = sideLengthA;
            this.sideLengthB = sideLengthB;
            this.sideLengthC = sideLengthC;
            this.sideLengthD = sideLengthD;
        }

        public virtual float CalculatePerimeter()
        {
            return sideLengthA + sideLengthB + sideLengthC + sideLengthD;
        }
        public virtual void PrintPerimeter(float perimeter)
        {
            Console.WriteLine("The perimeter of the Quadrangle is: " + perimeter);
        }

        public virtual void PrintArea(float area)
        {
            Console.WriteLine("The Area of the Quadrangle is: " + area);
        }
    }
}

File: RelationExampleC++/RelationExampleC++/Quadrangle.cpp
#include "Quadrangle.h"


Quadrangle::Quadrangle(float sideLengthA, float sideLengthB, float sideLengthC, float sideLengthD)
	: m_sideLengthA(sideLengthA), m_sideLengthB(sideLengthB), m_sideLengthC(sideLengthC), m_sideLengthD(sideLengthD)
{}

float Quadrangle::CalculatePerimeter()
{
	return  m_sideLengthA + m_sideLengthB + m_sideLengthC + m_sideLengthD;
}

void Quadrangle::PrintPerimeter(float perimeter)
{
	std::cout << "The perimeter of the Quadrangle is: " << perimeter << "\n";
}
void Quadrangle::PrintArea(float area)
{
	std::cout << "The Area of the Quadrangle is: " << area << "\n";
}